runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/GroupIntoBatchesOverride.java

```diff
@@ -210,20 +211,23 @@ public StreamingGroupIntoBatchesWithShardedKey(
     }
   }
 
-  private static <K, V> PCollection<KV<ShardedKey<K>, V>> ShardKeys(PCollection<KV<K, V>> input) {
+  private static <K, V> PCollection<KV<ShardedKey<K>, V>> shardKeys(PCollection<KV<K, V>> input) {
     KvCoder<K, V> inputCoder = (KvCoder<K, V>) input.getCoder();
     org.apache.beam.sdk.coders.Coder<K> keyCoder =
         (org.apache.beam.sdk.coders.Coder<K>) inputCoder.getCoderArguments().get(0);
     org.apache.beam.sdk.coders.Coder<V> valueCoder =
         (org.apache.beam.sdk.coders.Coder<V>) inputCoder.getCoderArguments().get(1);
     return input
         .apply(
-            "ShardKeys",
+            "Shard Keys",
             MapElements.via(
                 new SimpleFunction<KV<K, V>, KV<ShardedKey<K>, V>>() {
                   @Override
                   public KV<ShardedKey<K>, V> apply(KV<K, V> input) {
-                    return KV.of(ShardedKey.of(input.getKey(), new byte[0]), input.getValue());
+                    long tid = Thread.currentThread().getId();
```

Comment:
Thanks. These are likely to be re-used for all workers. Can you add in a statically initialized (for each worker) random long as well?

Reply:
Hmm I am a bit confused. The statically initialized random is at `GroupIntoBatches` class level. Is that correct? Would it be initialized only once the transform is created in the user pipeline and shared across workers? In the unit test, I found that the tid was different almost every time `SimpleFunction.apply` was called while the static tid was the same all the time.

Reply:
I meant you could have a static-level long created by java.util.UUID or similar, plus the thread id queried here.

Reply:
Ah I see. Addressed this in the latest commit.